### PR TITLE
Take existing blob size into account when updating delta.

### PR DESCRIPTION
**Pull request by vadimberezniker**
Enabling compression might cause some stored sizes to be updated.

---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/backends/pebble_cache/pebble_cache.go [hunk 1]
```diff
@@ -1185,22 +1185,24 @@ func (p *PebbleCache) Writer(ctx context.Context, r *resource.ResourceName) (int
...
 			tracing.AddStringAttributeToCurrentSpan(ctx, "pebble.duplicate_write", "true")
+			sizeDelta = bytesWritten - existingMD.GetStoredSizeBytes()
 		}
 
 		err = db.Set(fileMetadataKey, protoBytes, &pebble.WriteOptions{Sync: false})
 		if err == nil {
 			partitionID := fileRecord.GetIsolation().GetPartitionId()
-			if !alreadyExists {
+			if sizeDelta != 0 {
 				p.sendSizeUpdate(partitionID, fileMetadataKey, bytesWritten)
```
Comment (maggie-lou) [inline enterprise/server/backends/pebble_cache/pebble_cache.go:1203]: Should this be sizeDelta

File: enterprise/server/backends/pebble_cache/pebble_cache.go [hunk 2]
```diff
@@ -1185,22 +1185,24 @@ func (p *PebbleCache) Writer(ctx context.Context, r *resource.ResourceName) (int
...
 			}
-			metrics.DiskCacheAddedFileSizeBytes.With(prometheus.Labels{metrics.CacheNameLabel: p.name}).Observe(float64(bytesWritten))
+			metrics.DiskCacheAddedFileSizeBytes.With(prometheus.Labels{metrics.CacheNameLabel: p.name}).Observe(float64(sizeDelta))
```
Comment (maggie-lou) [inline enterprise/server/backends/pebble_cache/pebble_cache.go:1205]: Is it intended that this metric could have a negative value? Not sure whether this metric would be more useful if we used sizeDelta to show that we are overwriting files to be smaller, or if we used bytesWritten if we wanted to know the actual sizes of files on the disk

Comment (vadimberezniker) [reply enterprise/server/backends/pebble_cache/pebble_cache.go]: This is supposed to be bytesWritten. I changed the wrong line :facepalm:

File: enterprise/server/backends/pebble_cache/pebble_cache.go (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/server/backends/pebble_cache/pebble_cache.go]: Yeah :facepalm:

**Review by maggie-lou — APPROVED** [review]

**Pull request merged**